projects/storefrontapp-e2e-cypress/cypress/helpers/anonymous-consents.ts

```diff
@@ -0,0 +1,254 @@
+  cy.get('.cx-toggle-text').each($match => {
+    cy.wrap($match).should('contain.text', 'OFF');
+  });
+}
+
+export function rejectAllConsent() {
+  cy.get(`${ANONYMOUS_DIALOG} .cx-action-link`)
+    .contains('Reject All')
+    .click({ force: true });
+}
+
+export function acceptAllConsent() {
+  cy.get(`${ANONYMOUS_DIALOG} .cx-action-link`)
+    .contains('Allow All')
+    .click({ force: true });
+}
+
+// export function loggedInUserBannerTest() {
+//   it('should not render the banner', () => {
+//     cy.get(ANONYMOUS_BANNER).should('not.exist');
+//   });
+// }
+
+// export function loggedInUserFooterLinkTest() {
+//   it('should not render the footer link', () => {
+//     cy.get('.anonymous-consents').should('not.exist');
+//   });
+// }
+
+export function sessionLogin() {
+  standardUser.registrationData.email = generateMail(randomString(), true);
+  cy.requireLoggedIn(standardUser);
+}
+
+export function moveAnonymousUserToLoggedInUser() {
+  it('should ignore the anonymous consents and load the previously given registered consents', () => {
+    cy.selectUserMenuOption({
+      option: 'Consent Management',
+    });
+
+    giveConsent();
+    signOutUser();
+    checkBanner();
+
+    clickViewDetailsFromBanner();
+    giveOrRemoveAnonymousConsent(2);
+    closeDialog();
+
+    const loginPage = waitForPage('/login', 'getLoginPage');
+    cy.getByText(/Sign in \/ Register/i).click();
+    cy.wait(`@${loginPage}`);
+
+    login(
+      standardUser.registrationData.email,
+      standardUser.registrationData.password
+    );
+    checkBanner();
+
+    cy.selectUserMenuOption({
+      option: 'Consent Management',
+    });
+
+    cy.get('input[type="checkbox"]')
+      .first()
+      .should('be.checked');
+  });
+}
+
+export function movingFromAnonymousToRegisteredUser() {
+  it('should transfer anonoymous consents when registered', () => {
+    openDialogUsingFooterLink();
+    giveOrRemoveAnonymousConsent(2);
+    closeDialog();
+
+    registerUserAndCheckMyAccountConsent(false, 2);
+  });
+}
+
+// export function movingFromLoggedInUserToAnonymousUser() {
+//   it('should restore anonoymous consents when logging out', () => {
+//     const newUser = registerUser();
+//     cy.visit(PAGE_URL_LOGIN);
+//     login(newUser.email, newUser.password);
+//     signOutUser();
+
+//     openDialogUsingFooterLink();
+//     giveOrRemoveAnonymousConsent(2);
+//     closeDialog();
+
+//     cy.visit(PAGE_URL_LOGIN);
+//     login(newUser.email, newUser.password);
+//     signOutUser();
+
+//     openDialogUsingFooterLink();
+//     cy.get('.cx-toggle-text:nth-child(2)').should('contain.text', 'ON');
+//   });
+// }
+
+export function changeLanguageTest() {
+  it('should pull the new consent templates but preserve the consents state', () => {
+    openDialogUsingFooterLink();
+    acceptAllConsent();
+
+    siteContextChange('/', LANGUAGES, LANGUAGE_DE, LANGUAGE_LABEL);
+    openDialogUsingFooterLink();
+    checkConsentsGiven();
```

Comment:
Should we maybe also check that the German consent templates were pulled? Is this done in other e2e tests?